### PR TITLE
Remove LegacyESVersion.V_6_1_x constants

### LINKED ISSUE #1679: Remove LegacyESVersion.V_6_1_x constants
In prep for supporting Lucene 9, this issue tracks the removal of the now defunct `LegacyESVersion.V_6_1_*` along w/ all deprecated features associated with this version.

### PR DESCRIPTION
**Pull request by nknize**
This PR removes `LegacyESVersion.V_6_1_x` constants including all
pre-release versions and bug fixes.

closes #1679

### CODE REVIEW

**Review by adnapibar — APPROVED** [review]
LGTM - couple of nits

File: modules/percolator/src/test/java/org/opensearch/percolator/CandidateQueryTests.java [hunk 1]
```diff
@@ -673,7 +673,8 @@ public void testRangeQueries() throws Exception {
         IndexSearcher shardSearcher = newSearcher(directoryReader);
         shardSearcher.setQueryCache(null);
 
-        Version v = LegacyESVersion.V_6_1_0;
+        Version v = VersionUtils.randomIndexCompatibleVersion(random());
+        ;
```
[the remark this answers is not in the document]
Comment (nknize) [reply modules/percolator/src/test/java/org/opensearch/percolator/CandidateQueryTests.java:677]: :+1:

File: server/src/main/java/org/opensearch/index/mapper/SeqNoFieldMapper.java [hunk 2]
```diff
@@ -215,8 +213,7 @@ public void postParse(ParseContext context) throws IOException {
         // we share the parent docs fields to ensure good compression
         SequenceIDFields seqID = context.seqID();
         assert seqID != null;
-        final Version versionCreated = context.mapperService().getIndexSettings().getIndexVersionCreated();
-        final boolean includePrimaryTerm = versionCreated.before(LegacyESVersion.V_6_1_0);
+        final boolean includePrimaryTerm = false;
```
[the remark this answers is not in the document]
Comment (nknize) [reply server/src/main/java/org/opensearch/index/mapper/SeqNoFieldMapper.java:216]: :+1:

**Comment by nknize** [comment]
Unrelated failure (I could not reproduce this locally). This occurred twice:
```
./gradlew ':server:test' --tests "org.opensearch.cluster.routing.allocation.decider.ConcurrentRecoveriesAllocationDeciderTests.testClusterConcurrentRecoveries" -Dtests.seed=1E6DBFB47C410A54 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=ar-SY -Dtests.timezone=Asia/Ust-Nera -Druntime.java=17
```

```
org.opensearch.cluster.routing.allocation.decider.ConcurrentRecoveriesAllocationDeciderTests > testClusterConcurrentRecoveries FAILED
    java.lang.AssertionError: 
    Expected: <40>
         but: was <36>
        at __randomizedtesting.SeedInfo.seed([1E6DBFB47C410A54:C483ECD37CB36683]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.opensearch.cluster.routing.allocation.decider.ConcurrentRecoveriesAllocationDeciderTests.testClusterConcurrentRecoveries(ConcurrentRecoveriesAllocationDeciderTests.java:221)
```

/cc @jainankitk

**Comment by nknize** [comment]
start gradle check

**Comment by nknize** [comment]
HAProxy issue:

```
./gradlew ':qa:remote-clusters:integTest' --tests "org.opensearch.cluster.remote.test.RemoteClustersIT.testHAProxyModeConnectionWorks" -Dtests.seed=4C426D008FCAC0C0 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=hu -Dtests.timezone=America/Louisville -Druntime.java=17
```

```
  2> java.lang.AssertionError
        at __randomizedtesting.SeedInfo.seed([4C426D008FCAC0C0:4B34B8CBB44CB69D]:0)
        at org.junit.Assert.fail(Assert.java:86)
        at org.junit.Assert.assertTrue(Assert.java:41)
        at org.junit.Assert.assertTrue(Assert.java:52)
        at org.opensearch.cluster.remote.test.RemoteClustersIT.testHAProxyModeConnectionWorks(RemoteClustersIT.java:125)
  2> NOTE: leaving temporary files on disk at: /var/CITOOL/workflow/OpenSearch_CI/PR_Checks/Gradle_Check/search/qa/remote-clusters/build/testrun/integTest/temp/org.opensearch.cluster.remote.test.RemoteClustersIT_4C426D008FCAC0C0-001
  2> NOTE: test params are: codec=Asserting(Lucene87): {}, docValues:{}, maxPointsInLeafNode=878, maxMBSortInHeap=5.294271434390409, sim=Asserting(RandomSimilarity(queryNorm=false): {}), locale=hu, timezone=America/Louisville
  2> NOTE: Linux 5.4.0-1045-aws amd64/Eclipse Adoptium 17.0.1 (64-bit)/cpus=72,threads=1,free=436388792,total=536870912
  2> NOTE: All tests run in this JVM: [RemoteClustersIT]
```

**Comment by nknize** [comment]
start gradle check

**Pull request merged**